File: tests/fixtures/edgecase_repo/csharp/Service1.cs
using System;

namespace Example.Gen
{
    public interface IHandler1
    {
        bool Handle(string payload);
    }

    public class Base1
    {
        protected string Name;
    }

    public class Service1 : Base1, IHandler1
    {
        public Service1(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}

File: tests/fixtures/edgecase_repo/scala/Service1.scala
package com.example.gen

import com.example.gen.Base1

trait Handler1 {
  def handle(payload: String): Boolean
}

class Base1(val name: String)

class Service1(name: String) extends Base1(name) with Handler1 {
  def handle(payload: String): Boolean = {
    this.validate(payload)
    true
  }

  private def validate(payload: String): Unit = {}
}

object Helpers1 {
  def helper(value: String): String = value
}
